### ISSUE TITLE
Order slain Automatic tickbox not functioning correctly

### ISSUE THREAD

**Issue by korzxd** [dixhuit/unitcrunch-issue-tracker#97]
### Debug data
I have sent Debug Data.

### Describe the bug
In an Advanced Profile with the Order slain Automatic tickbox enabled, then when adding or creating Units/Model Types the Order slain does not automatically change.

If you disable that Automatic checkbox and enable it, it will reconfigure the Order Slain correctly.

Another method was to save the profile then edit it to reopen it now displaying the correct Order Slain and then saving it again. This method will not work if you edit the profile then click Cancel instead of save, it will keep the Original order making the Leader Unit the first target.

### To Reproduce
1. Click **Manage Profiles**
2. Click **New Profile**
3. Click **Advanced**
4. Click **Create Unit**
5. Enable the **Leader Checkbox**
6. See Result in **Order Slain** Section

If you Click **Create Unit** again you'll see the Leader Unit will now be in the middle rather than at the bottom.

### Expected behavior
The Order Slain would update as Units or Model Types are created, deleted or made a Leader. A different expectation could be the Order Slain updating on clicking Save.

**Comment by dixhuit** [comment]
Thanks for this. Will investigate ASAP.

**Comment by dixhuit** [comment]
Your description and your reproduction don't match. The reproduction does reproduce the bug, but that is specifically because the "Leader" checkbox on any unit is not triggering re-evaluation of automatic order slain. Adding units and/or model types does trigger re-evaluation of auto order slain. Just wanted to check that we're seeing the same thing.

So I need to watch the "Leader" checkbox on all units as that may influence order slain when set to Automatic. Similarly, I need to also watch for any changes to any model type's "Model count" as that is also part of the automatic order slain algorithm.

**Comment by korzxd** [comment]
> specifically because the "Leader" checkbox on any unit is not triggering re-evaluation of automatic order slain.

I agree the Leader checkbox is not triggering any re-evaluation.

> Adding units and/or model types does trigger re-evaluation of auto order slain.

I don't quite know if i understand what you mean with this part, I've found it just keeps placing any newly created unit and/or model types at position 1 and pushes the rest down.

I quickly created this Advanced Profile to see if it helps, this was made in the following order.

1. The first Unit with it's First model type i set the characteristics then named it "A: 5 Models, 1 Wound, fodder".
2. I then created a new model type within that unit set the characteristics then named it "B: 1 Model, 2 Wound, sergeant"
3. Then i created a new Unit set it's characteristics, ticked Leader then named it "C: Leader"
4. Lastly i made another new Unit set it's characteristics then named it "D: 1 Model, 5 Wounds, Leader2"

![image](https://github.com/user-attachments/assets/1b6399fe-5e38-4503-abce-fe76e04623b3)
In this circumstance i would've expected the auto order slain to re-evaluate when "Adding units and/or model types" but none of that happened it just kept placing the most recently created one at position 1.

When i Disabled then Enabled the Automatic tickbox it did re-evaluate the order as seen below.
![image](https://github.com/user-attachments/assets/0cc2a9eb-8b82-4565-8863-50eed64ad4d7)

After that i edited the characteristics of every Model type and there was no update.

I hope this somewhat helps 😅
